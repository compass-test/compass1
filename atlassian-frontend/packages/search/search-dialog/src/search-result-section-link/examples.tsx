import React from 'react';
import { SearchResultSectionLink } from './search-result-section-link';
import { action } from '@storybook/addon-actions';

const DummyLinkComponent: React.FunctionComponent = ({ children, ...rest }) => (
  <a
    {...rest}
    onClick={(e) => {
      e.preventDefault();
      action('Linkcomponent clicked')(e);
    }}
  >
    {children}
  </a>
);

export const Basic = () => (
  <SearchResultSectionLink href={`https://www.atlassian.com/${Math.random()}`}>
    Some link
  </SearchResultSectionLink>
);

export const WithLinkComponent = () => (
  <SearchResultSectionLink linkComponent={DummyLinkComponent} href={`#`}>
    Some link with Link Component
  </SearchResultSectionLink>
);

export default {
  title: 'Search Dialog/Search Result Section Link',
};
