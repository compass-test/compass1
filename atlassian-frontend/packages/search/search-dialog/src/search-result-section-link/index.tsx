export { SearchResultSectionLink } from './search-result-section-link';
export type { Props as SearchResultSectionLinkProps } from './search-result-section-link';
