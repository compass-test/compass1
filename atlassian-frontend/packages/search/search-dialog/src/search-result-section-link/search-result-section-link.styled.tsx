import Button from '@atlaskit/button';
import styled from '@emotion/styled';
import { grid } from '../style-utils';
import { B400 } from '@atlaskit/theme/colors';

// The margin is reduced to account for the Button component's 8px internal padding
// such that it aligns with the SearchResult icon.
// We need the `&&&` as Confluence has some CSS that overrides the styling otherwise
export const SectionLink = styled(Button)`
  margin-left: ${grid.half().px};
  color: ${B400};
  cursor: pointer !important;
  &:hover {
    text-decoration: underline !important;
  }
`;
