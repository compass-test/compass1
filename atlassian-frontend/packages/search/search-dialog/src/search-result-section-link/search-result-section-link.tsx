import React from 'react';
import { ButtonProps } from '@atlaskit/button';
import { SectionLink } from './search-result-section-link.styled';

export interface Props extends ButtonProps {
  /**
   * This will replace all links with the given component. This will be used to ensure that results provided will be properly SPA transitioned.
   * Do not use this for styling.
   */
  linkComponent?: React.ElementType<any>;
  /**
   * An HTML id that will be set on the underlying HTML element.
   */
  id?: string;
}

export const SearchResultSectionLink: React.FunctionComponent<Props> = ({
  children,
  linkComponent,
  ...rest
}) => (
  /* component must either be `undefined` or an component. A null or false will throw an error */
  <SectionLink
    appearance="link"
    {...rest}
    component={linkComponent || undefined}
  >
    {children}
  </SectionLink>
);
