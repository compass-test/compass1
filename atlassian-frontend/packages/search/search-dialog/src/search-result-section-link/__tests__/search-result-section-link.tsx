import React from 'react';
import { shallow } from 'enzyme';
import { SearchResultSectionLink } from '../search-result-section-link';

describe('<SearchResultSectionLink />', () => {
  it('SearchResult matches snapshot', () => {
    const wrapper = shallow(
      <SearchResultSectionLink href={'http://google.com'} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
