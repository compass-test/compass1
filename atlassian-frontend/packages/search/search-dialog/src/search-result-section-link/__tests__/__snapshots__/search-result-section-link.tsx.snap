// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultSectionLink /> SearchResult matches snapshot 1`] = `
<Styled(Button)
  appearance="link"
  href="http://google.com"
/>
`;
